### DISCUSSION TITLE
Open all sidebar panels when any are opened?

### DISCUSSION THREAD

**Discussion by mrjones2014** [folke/edgy.nvim#16] [Q&A] [answered]
I basically want this layout. I'm using the command `Neotree` to open it, but then I have to manually navigate to the other 2 windows for them to be expanded. How can I have all the windows in that sidebar expanded by default?

![image](https://github.com/folke/edgy.nvim/assets/8648891/4284ee17-df1b-407d-be4f-97aa8b2b5925)

My config: 

```lua
return {
  'folke/edgy.nvim',
  dependencies = {
    {
      'nvim-neo-tree/neo-tree.nvim',
      config = function()
        require('neo-tree').setup({
          open_files_do_not_replace_types = { 'terminal', 'Trouble', 'qf', 'edgy' },
        })
      end,
    },
    {
      'simrat39/symbols-outline.nvim',
      config = function()
        require('symbols-outline').setup()
      end,
    },
  },
  event = 'VeryLazy',
  config = function()
    require('edgy').setup({
      exit_when_last = true,
      animate = {
        enabled = false,
      },
      options = {
        right = { size = 30 },
      },
      keys = {
        ['q'] = function(win)
          win.view.edgebar:close()
        end,
      },
      right = {
        {
          title = 'Neo-Tree Buffers',
          ft = 'neo-tree',
          filter = function(buf)
            return vim.b[buf].neo_tree_source == 'buffers'
          end,
          pinned = true,
          open = 'Neotree position=top buffers',
          size = { height = 0.1 },
        },
        {
          title = 'Neo-Tree',
          ft = 'neo-tree',
          pinned = true,
          filter = function(buf)
            return vim.b[buf].neo_tree_source == 'filesystem'
          end,
          size = { height = 0.5 },
        },
        {
          ft = 'Outline',
          pinned = true,
          open = 'SymbolsOutline',
        },
      },
    })
  end,
}
```

**Comment by mrjones2014** [comment]
Also it seems like the windows are not resizable?

**Reply by folke** [reply]
Added a section of limitations to the readme https://github.com/folke/edgy.nvim#%EF%B8%8F-limitations

**Comment by folke** [comment, marked as answer]
You can now call `require("edgy").open(pos?)` to open all the pinned views in a given edgebar or all edgebars.

**Reply by mrjones2014** [reply]
Thanks!

**Reply by mrjones2014** [reply]
This doesn't seem to work quite properly for me. It's possible that it's an issue with `simrat39/symbols-outline.nvim`, but I get the outline window not opening until I navigate into it still. The `Neotree` file tree and buffers window do open properly. My sidebars are configured like so: 

```lua
right = {
  {
    title = 'Neo-Tree Buffers',
    ft = 'neo-tree',
    filter = function(buf)
      return vim.b[buf].neo_tree_source == 'buffers'
    end,
    pinned = true,
    open = 'Neotree position=top buffers',
    size = { height = 0.1 },
  },
  {
    title = 'Neo-Tree',
    ft = 'neo-tree',
    filter = function(buf)
      return vim.b[buf].neo_tree_source == 'filesystem'
    end,
    size = { height = 0.5 },
  },
  {
    ft = 'Outline',
    pinned = true,
    open = 'SymbolsOutline',
  },
},
```

and I'm opening it with a keymap that does:

```lua
function()
  vim.cmd.Neotree()
  require('edgy').open('right')
end
```

**Reply by folke** [reply]
For outline to open properly, make sure you start from a real buffer that has an lsp attached.
The `open` method will actually move the cursor back to the last main editor window before triggering the open command.

The problem may be that neotree opens, and gets focus. Neo-tree only sets the source type later, so maybe that's set after outline starts to open. So this means that it will be treated as a main window, till its filetype is set.

Check if it works when opening neotree manually and then do that edgy open manually from the cmdline

**Reply by mrjones2014** [reply]
> For outline to open properly, make sure you start from a real buffer that has an lsp attached.

I am.

> Check if it works when opening neotree manually and then do that edgy open manually from the cmdline

When I do this, _neither_ the buffers window or the outline window open.

**Reply by folke** [reply]
What do you mean?

I just meant, open neotree with a keymap or with `:`, and then open edgy pinned windows with the cmdline like this
![image](https://github.com/folke/edgy.nvim/assets/292349/bc74b347-185c-47a9-ae78-eb3d965e064e)

**Reply by mrjones2014** [reply]
> I just meant, open neotree with a keymap or with :, and then open edgy pinned windows with the cmdline like this

Yeah, when I do that, I end up with this, top and bottom windows are collapsed.

![image](https://github.com/folke/edgy.nvim/assets/8648891/e269fc12-24b3-4841-b67a-e4a81d5e2595)

**Reply by folke** [reply]
You're right. I just pushed a fix to open pinned views when the sidebar is already open. That probably fixes your other issue too

**Reply by mrjones2014** [reply]
It sorta works but now I get an infinite loop of these errors 
![image](https://github.com/folke/edgy.nvim/assets/8648891/4bddb6b5-e644-43cc-8922-bbc6d896723f)

**Reply by mrjones2014** [reply]
Actually I can't reproduce after restarting nvim. Seems to work now!

**Reply by Rukenshia** [reply]
> You can now call `require("edgy").open(pos?)` to open all the pinned views in a given edgebar or all edgebars.

@folke Is there a way to expand/open a window without focussing it? I want to expand my left sidebar when opening neovim, but focus is always taken if I use `require('edgy').open('left')`. I'm also using auto-session to pick up where I left off, so being able to expand the windows without having to then switch windows again would be nice

**Reply by folke** [reply]
@Rukenshia that's not edgt taking that focus. Those windows are doing that. Nothing I can do about this.

**Reply by Rukenshia** [reply]
> @Rukenshia that's not edgt taking that focus. Those windows are doing that. Nothing I can do about this.

Alright, thank you! Appreciate the instant reply!

**Reply by folke** [reply]
Wait, I actually may be wrong here. When re-organizing the windows, that may take focus as well
